include/zephyr/bluetooth/mesh/access.h

```diff
@@ -1129,23 +1129,31 @@ int bt_mesh_model_correspond(const struct bt_mesh_model *corresponding_mod,
  */
 bool bt_mesh_model_is_extended(const struct bt_mesh_model *model);
 
-/** @brief Indicate that the composition data will change on next bootup.
+/** @brief Set Composition Data Page contents.
  *
- *  Tell the config server that the composition data is expected to change on
- *  the next bootup, and the current composition data should be backed up.
+ *  Can only be used for pages >= 128. If this is called with a page number
+ *  not present on the device, it will return an error code.
  *
- *  @return Zero on success or (negative) error code otherwise.
+ *  @param page The page to set the contents of. Must be >= 128.
+ *  @param data Pointer to new page contents.
+ *  @param len Length of new page contents.
+ *
+ *  @retval 0 on success, (negative) error code otherwise.
  */
-int bt_mesh_comp_change_prepare(void);
```

Comment:
May be nice to have removal of old stuff as its own commit at least? Easier to discern then what is removal and what is modification, and if some deprecation log etc. needs to be added, it can be contained in the removal commit.

Reply:
That does not make much sense either I think? Either the removal-commit needs to happen first, but then, what should the models do? Just not work until the follow-up commit with the new API shows up? Or, the removal commit comes last, but then, what should these functions do before being removed in the second commit? Nothing?

Reply:
If removal first modal functionality isn't given, I guess it is fine as is, or after. Just was confused at some points regarding changes--which turned out to be removals + new functions. Plus smallest reasonably possible commits means easy to merge in small chunks and easier to review :)

Reply:
Basically, we need to
a) remove old API and
b) add new API

Splitting this in two commits, we either have 2 APIs or 0 APIs in the middle, neither of which make sense. I see your point about the diff being confusing in places, but I don't see a good way to avoid it, sadly

Reply:
Try to remove it without deprecation. We may have to write something in the migration doc.